newsrevenuehub/rev-engine
# DEV-1578 support distinct role types in the spa

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

#### What's this PR do?

Changes to SPA (and recursive changes to API that emerged) to enable role based permissions.

One consequence of this PR is that page, template, style, and contribution data will be filtered via user roles.

Some noteworthy code changes:

*Backend*

- Simplifies API endpoint configuration for creation, such that there are not query parameters being passed to endpoints that otherwise expect a body of data (aka, no longer faceting based on RP and ORG slugs)
- Adds `apps.common.fields.SerializedOnReadElsePk` field which can be used to enable an API resource where a field is PK on creation, otherwise serialized
- Cleans up some use of custom mixins that were not rquired in views
- Adds some fields to OrganizationInlineSerializer to make front end work
- Adds a validator (`RpPkIsForOwnedRp`) which can be used to validate if the requesting user should be able to reference a related revenue program

*SPA*

- Adds a `forceLogin` command that can be used to directly force login without going through sign on page flow in tests, and replaces a bunch of current use of cy.login with this new method.
- Removes unused revenue program route
- Renames `OrganizationContext` to `PaymentProviderContext`
- In `AddPageModal`, the list of revenue programs that appear in the dropdown now comes from the list of available RPs in a user's role information instead of pulled from API resource.
- Removes unused src/components/organization/Organization.js


#### Why are we doing this? How does it help us?

will allow us to ship role-based permissions

#### How should this be manually tested?

Run the dev server locally (preferably after retrieving a recent back up from staging and restoring it, then running migrations). Log in to the SPA as a hub admin or super user and you should see all content. Log in as an org or rp user, and you should only see content that you own.

#### Have automated unit tests been added?

Added and edited.

#### How should this change be communicated to end users?

#### Are there any smells or added technical debt to note?


#### Has this been documented? If so, where?

#### What are the relevant tickets?
**(Note: Please use syntax [JIRA-123] to link any relevant tickets)**

[DEV-1578]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them?

Revisit DEV-699 once this is merged into it, and do clean up before submitting that PR for final review.
